rfcs/text/0000-rfc-library-preparation.md

```diff
@@ -0,0 +1,428 @@
+    "description": "A [project-wide] unique ID for the biomaterial.",
+},
+"biomaterial_name": {
+    "description": "A short, descriptive name for the biomaterial that need not be unique.",
+},
+"biomaterial_description": {
+    "description": "A general description of the biomaterial.",
+},
+"ncbi_taxon_id" : {
+    "description": "A taxonomy ID (taxonID) from NCBI.",
+},
+"biosamples_accession": {
+    "description": "A BioSamples accession.",
+},
+"insdc_sample_accession": {
+    "description": "An International Nucleotide Sequence Database Collaboration (INSDC) sample accession.",
+},
+"supplementary_files": {
+    "description": "A list of filenames of biomaterial-level supplementary files.",
+},
+"genotype*": {
+    "description": "Genotype of the biomaterial.",
+},
+"HDBR_accession*": {
+    "description": "A Human Developmental Biology Resource (HDBR) sample accession.",
+}
+```
+> *Consider removing `genotype` and `HDBR_accession` from core as they don't make sense here and in other biomaterials.
+
+And the `provenance.json` schema contains the following properties (all supplied by Ingestion Service):
+
+```
+"describedBy":  {
+    "description": "The URL reference to the schema.",
+},
+"schema_version": {
+    "description": "The version number of the schema in major.minor.patch format.",
+},
+"submission_date": {
+    "description": "When project was first submitted to database.",
+},
+"submitter_id": {
+    "description": "ID of individual who first submitted project.",
+},
+"update_date": {
+    "description": "When project was last updated.",
+},
+"updater_id": {
+    "description": "ID of individual who last updated project.",
+},
+"document_id": {
+    "description": "Identifier for document.",
+},
+"accession": {
+    "description": "A unique accession for this entity, provided by the broker.",
+}
+```
+
+The new `library_preparation.biomaterial_core.biomaterial_id` field will replace the function of the current `sequence_file.library_prep_id` field in the sequence file schema. Data contributors will be required to provide this field to ensure the correct experimental design is represented. It is unclear how the requirement will be enforced: at the moment there is no validation for the presence of a particular JSON type in a logical unit (i.e. bundle).
+
+The required fields for the library preparation entity will be:
+- `library_preparation.biomaterial_core.biomaterial_id`
+- `library_preparation.biomaterial_core.ncbi_taxon_id`
+
+#### Added/changed library preparation-specific fields
+
+The new `library_preparation.json` schema will include a field for capturing INSDC experiment accessions if the project is already archived. This field (`process.insdc_experiment.insdc_experiment_accession`) generally represents a single library preparation in the archives (ENA, SRA, DDJB). This field currently lives awkwardly in a process module for historical reasons and should be moved to the new library preparation schema. 
```

Comment:
If the experiment has been submitted to one of the mentioned archives, then it will necessarily have a SRX/ERX/DRX accession, AFAIK. Of course, some submitted data might not be in an archive yet, in which case it won't have this ID.